src/spyglass/decoding/v0/sorted_spikes.py

```diff
@@ -0,0 +1,451 @@
+speeds. eLife 10, e64505 (2021).
+
+"""
+from spyglass.utils import logger
+
+try:
+    import pprint
+
+    import datajoint as dj
+    import numpy as np
+    import pandas as pd
+    from replay_trajectory_classification.classifier import (
+        _DEFAULT_CONTINUOUS_TRANSITIONS,
+        _DEFAULT_ENVIRONMENT,
+        _DEFAULT_SORTED_SPIKES_MODEL_KWARGS,
+    )
+    from replay_trajectory_classification.discrete_state_transitions import (
+        DiagonalDiscrete,
+    )
+    from replay_trajectory_classification.initial_conditions import (
+        UniformInitialConditions,
+    )
+
+    from spyglass.common.common_behav import (
+        convert_epoch_interval_name_to_position_interval_name,
+    )
+    from spyglass.common.common_interval import IntervalList
+    from spyglass.common.common_nwbfile import AnalysisNwbfile
+    from spyglass.common.common_position import IntervalPositionInfo
+    from spyglass.decoding.core import (
+        convert_valid_times_to_slice,
+        get_valid_ephys_position_times_by_epoch,
+    )
+    from spyglass.decoding.dj_decoder_conversion import (
+        convert_classes_to_dict,
+        restore_classes,
+    )
+    from spyglass.spikesorting.spikesorting_curation import CuratedSpikeSorting
+    from spyglass.utils import SpyglassMixin, logger
+
+    schema = dj.schema("decoding_sortedspikes")
+
+    @schema
+    class SortedSpikesIndicatorSelection(SpyglassMixin, dj.Lookup):
+        """Bins spike times into regular intervals given by the sampling rate.
+        Start and stop time of the interval are defined by the interval list.
+        """
+
+        definition = """
+        -> CuratedSpikeSorting
+        -> IntervalList
+        sampling_rate=500 : float
+        ---
+        """
+
+    @schema
+    class SortedSpikesIndicator(SpyglassMixin, dj.Computed):
+        """Bins spike times into regular intervals given by the sampling rate.
+        Useful for GLMs and for decoding.
+
+        """
+
+        definition = """
+        -> SortedSpikesIndicatorSelection
+        ---
+        -> AnalysisNwbfile
+        spike_indicator_object_id: varchar(40)
+        """
+
+        def make(self, key):
+            pprint.pprint(key)
+            # TODO: intersection of sort interval and interval list
+            interval_times = (IntervalList & key).fetch1("valid_times")
+
+            sampling_rate = (SortedSpikesIndicatorSelection & key).fetch(
+                "sampling_rate"
+            )
+
+            time = self.get_time_bins_from_interval(
+                interval_times, sampling_rate
+            )
+
+            spikes_nwb = (CuratedSpikeSorting & key).fetch_nwb()
+            # restrict to cases with units
+            spikes_nwb = [entry for entry in spikes_nwb if "units" in entry]
+            spike_times_list = [
+                np.asarray(n_trode["units"]["spike_times"])
+                for n_trode in spikes_nwb
+            ]
+            if len(spike_times_list) > 0:  # if units
```

Comment:
It looks like the remained of this func is only run if units, so i would flip the logic and return early to save indenting below